cmd/soroban-cli/src/config/address.rs

```diff
@@ -56,8 +61,43 @@ impl Address {
 
     pub fn resolve_secret(&self, locator: &locator::Args) -> Result<secret::Secret, Error> {
         match &self {
-            Address::MuxedAccount(muxed_account) => Err(Error::CannotSign(muxed_account.clone())),
             Address::AliasOrSecret(alias) => Ok(locator.read_identity(alias)?),
+            a @ Address::MuxedAccount(_) => {
+                Err(Error::CannotSign(a.resolve_muxed_account(locator, None)?))
+            }
         }
     }
 }
+
+#[derive(Clone, Debug)]
+pub struct KeyName(pub String);
+
+impl std::ops::Deref for KeyName {
+    type Target = str;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl std::str::FromStr for KeyName {
+    type Err = Error;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if !s.chars().all(allowed_char) {
+            return Err(Error::InvalidKeyName(s.to_string()));
+        }
+        if s == "ledger" {
+            return Err(Error::InvalidKeyName(s.to_string()));
+        }
+        Ok(KeyName(s.to_string()))
```

Comment:
Since we are finally validating key names, should we also set a max length?

Reply:
oh yeah, i didn't think about that - what do you think the max length should be?

Reply:
So it looks like the most common max length for a filename is 255 characters. chars and length can differ if using some unicode characters that take up more than one byte. However, in this case we are limiting the requirement to ascii characters. We also use `.toml`. Though I'm not sure if the extension counts as part of the filepath's length. But let's just make it 250 just in case. That is plenty!

Reply:
sounds good, i'll push a fix up shortly!